include/exadg/solvers_and_preconditioners/multigrid/coarse_grid_solvers.h

```diff
@@ -318,9 +324,12 @@ class MGCoarseAMG
   typedef LinearAlgebra::distributed::Vector<typename Operator::value_type> VectorTypeMultigrid;
 
 public:
-  MGCoarseAMG(Operator const & op, AMGData data = AMGData())
+  MGCoarseAMG(Operator const & op, bool const use_boomer_amg, AMGData data = AMGData())
   {
-    amg_preconditioner.reset(new PreconditionerAMG<Operator, TrilinosNumber>(op, data));
+    if(use_boomer_amg)
+      amg_preconditioner.reset(new PreconditionerBoomerAMG<Operator, TrilinosNumber>(op, data));
```

Comment:
please rename into `NumberAMG` and `VectorTypeAMG` (should only affect this file).